Dockerfile.ocp

```diff
@@ -0,0 +1,27 @@
+ARG BUILDER_IMAGE=brew.registry.redhat.io/rh-osbs/openshift-golang-builder:rhel_9_1.23
+ARG BASE_IMAGE=registry.access.redhat.com/ubi9/ubi-micro:latest
+
+
+# Build the manager binary
+FROM ${BUILDER_IMAGE} AS builder
+FROM --platform=${BUILDPLATFORM} ${BUILDER_IMAGE} as builder
+
+ARG TARGETOS
+ARG TARGETARCH
+
+WORKDIR /workspace
+COPY . .
+
+# Build
+# the GOARCH has not a default value to allow the binary be built according to the host where the command
+# was called. For example, if we call make docker-build in a local env which has the Apple Silicon M1 SO
+# the docker BUILDPLATFORM arg will be linux/arm64 when for Apple x86 it will be linux/amd64. Therefore,
+# by leaving it empty we can ensure that the container and binary shipped on it will have the same platform.
+RUN CGO_ENABLED=${CGO_ENABLED:-1} GOOS=${TARGETOS:-linux} GOARCH=${TARGETARCH} go build -a -o manager cmd/main.go
```

Comment:
I'm wondering whether it is possible to use build-machinery-go like in https://github.com/openshift/cli-manager/blob/5c0184d6697a2d4e40f9610b763943549525fac8/Makefile#L14-L17 to manage all these things for us. But I understand that this is fork of upstream, if it is not possible.

Reply:
I'd rather not change this too much.